src/hooks/protections/received_event.cpp

```diff
@@ -738,6 +738,34 @@ namespace big
 			}
 			break;
 		}
+		case eNetworkEvents::NETWORK_PTFX_EVENT:
+		{
+			if (!plyr)
+			{ // If there is no player its likely a modded event so just ack it
+				g_pointers->m_gta.m_send_event_ack(event_manager, source_player, target_player, event_index, event_handled_bitset);
+				return;
+			}
+			if (g.protections.ptfx_spam_protection)
+			{
+				if (plyr->block_ptfx)
+				{
+					g_pointers->m_gta.m_send_event_ack(event_manager, source_player, target_player, event_index, event_handled_bitset);
+					return;
+				}
+				plyr->m_ptfx_ratelimit.process();
+				if (plyr->m_ptfx_ratelimit.exceeded_last_process())
+				{
+					if (!plyr->ptfx_spam_notification_sent)
+					{
+						plyr->ptfx_spam_notification_sent = true;
+						g.reactions.ptfx_spam.process(plyr);
+					}
+					g_pointers->m_gta.m_send_event_ack(event_manager, source_player, target_player, event_index, event_handled_bitset);
+					return;
+				}

```

Comment:
use ratelimit logic similar to `m_play_sound_rate_limit`. `exceeded_last_process` is to avoid notification spam it will only block ptfx once. For blocking every event put the code inside `process` block

Reply:
The goal isn;t to block every single PTFX event, You can do this per player, This prevents modded PTFX from being used to spam you so you can't play the game, An example of this a smoke screen PTFX that can make your screen fully black and you can not see anything. I will have a look at that tho

Reply:
I meant it for that player only. Not all.

Reply:
do this
```C++
			if (plyr && plyr->m_ptfx_ratelimit.process())
			{
				if (plyr->m_ptfx_ratelimit.exceeded_last_process())
					g.reactions.ptfx_spam.process(plyr);

				g_pointers->m_gta.m_send_event_ack(event_manager, source_player, target_player, event_index, event_handled_bitset);
				return;
			}
```

Reply:
> do this
> 
> ```c++
> 			if (plyr && plyr->m_ptfx_ratelimit.process())
> 			{
> 				if (plyr->m_ptfx_ratelimit.exceeded_last_process())
> 					g.reactions.ptfx_spam.process(plyr);
> 
> 				g_pointers->m_gta.m_send_event_ack(event_manager, source_player, target_player, event_index, event_handled_bitset);
> 				return;
> 			}
> ```

If the "Block PTFX" option is enabled per player it is not handled with your code ,secondly there is no need to rate limit it if its going to be blocked anyways.

Reply:
got it. The code you wrote is for notification purpose here. Clever coding.

Reply:
good to go. No need to change anything

Reply:
> good to go. No need to change anything

👍 just need to find what network event was removed, Make the new options fit better on the ui and this is ready to merge i think!

Reply:
You can ask around if you have doubt related to UI

Reply:
> You can ask around if you have doubt related to UI

I think i figured it out. Right now if you were to compile this the last column would be longer than the others so instead i moved them to be correctly sized